### ISSUE TITLE
Arista Doesn't Work With openSUSE

### ISSUE THREAD

**Issue by arnieswap** [danielgtaylor/arista#134]
When I try to transcode movies for Android Tablets using Galaxy Tab or Android Droid X preset, Arista throws error 'can't add item due to missing elements"

**Comment by danielgtaylor** [comment]
Please make sure you've installed the relevant GStreamer plugins, e.g.
gstreamer-plugins-bad/ugly and gstreamer-ffmpeg.

Take care,
Daniel

On Tue, Mar 20, 2012 at 20:29, arnieswap
reply@reply.github.com
wrote:

> When I try to transcode movies for Android Tablets using Galaxy Tab or Android Droid X preset, Arista throws error 'can't add item due to missing elements"
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/danielgtaylor/arista/issues/134

## 

Daniel G. Taylor
http://programmer-art.org/

**Comment by arnieswap** [comment]
Yes all those packages are installed. I installed codecs from [http://en.opensuse.org/Restricted_formats]

Output

muktware@linux-fdhk:~> arista-gtk
*\* Message: pygobject_register_sinkfunc is deprecated (GstObject)
arista.presets [433]: INFO Attempting to install elements: x264enc
*\* Message: PackageKit: xid = 0
*\* Message: PackageKit: Codec nice name: GStreamer element x264enc
*\* Message: PackageKit: structure: gstreamer0.10(element-x264enc)()(64bit)
*\* Message: PackageKit: Did not install codec: Message did not receive a reply (timeout by message bus)
arista.presets [442]: ERROR Unable to install required elements!
arista.presets [433]: INFO Attempting to install elements: x264enc
*\* Message: PackageKit: xid = 0
*\* Message: PackageKit: Codec nice name: GStreamer element x264enc
*\* Message: PackageKit: structure: gstreamer0.10(element-x264enc)()(64bit)
*\* Message: PackageKit: Did not install codec: Message did not receive a reply (timeout by message bus)
arista.presets [442]: ERROR Unable to install required elements!

The error it throws is 
Cannot add conversion to queue because of missing elements!

**Comment by danielgtaylor** [comment]
Does gst-inspect-0.10 x264enc show anything in the terminal? You need to
have the x264enc element installed for most of Arista's presets to work
properly. Perhaps that is missing from the system packages, or is renamed
or something?

Take care,
Daniel

On Fri, Mar 23, 2012 at 13:37, arnieswap <reply+i-3738507-095e44302a1ec677d
56ad35f9073b268979bba70-106826@reply.github.com> wrote:

> Yes all those packages are installed. I installed codecs from [
> http://en.opensuse.org/Restricted_formats]
> 
> Output
> 
> muktware@linux-fdhk:~> arista-gtk
> *\* Message: pygobject_register_sinkfunc is deprecated (GstObject)
> arista.presets [433]: INFO Attempting to install elements: x264enc
> *\* Message: PackageKit: xid = 0
> *\* Message: PackageKit: Codec nice name: GStreamer element x264enc
> *\* Message: PackageKit: structure: gstreamer0.10(element-x264enc)()(64bit)
> *\* Message: PackageKit: Did not install codec: Message did not receive a
> reply (timeout by message bus)
> arista.presets [442]: ERROR Unable to install required elements!
> arista.presets [433]: INFO Attempting to install elements: x264enc
> *\* Message: PackageKit: xid = 0
> *\* Message: PackageKit: Codec nice name: GStreamer element x264enc
> *\* Message: PackageKit: structure: gstreamer0.10(element-x264enc)()(64bit)
> *\* Message: PackageKit: Did not install codec: Message did not receive a
> reply (timeout by message bus)
> arista.presets [442]: ERROR Unable to install required elements!
> 
> The error it throws is
> Cannot add conversion to queue because of missing elements!
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/danielgtaylor/arista/issues/134#issuecomment-4664158

## 

Daniel G. Taylor
http://programmer-art.org/

**Comment by arnieswap** [comment]
gst-inspect-0.10 x264enc 

Output
No such element of plugin 'x264enc

Best wishes
Swapnil

**Comment by danielgtaylor** [comment]
I would file a bug with your distribution or ask what package you need to
install to get the x264enc gstreamer plugin. Without it you can't do all
that much with Arista.

Take care,
Daniel

On Fri, Mar 23, 2012 at 16:31, arnieswap <
reply@reply.github.com

> wrote:
> 
> gst-inspect-0.10 x264enc
> 
> Output
> No such element of plugin 'x264enc
> 
> Best wishes
> Swapnil
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/danielgtaylor/arista/issues/134#issuecomment-4667332

## 

Daniel G. Taylor
http://programmer-art.org/

**Comment by arnieswap** [comment]
@danielgtaylor Thanks. That will help us a lot. I will keep an eye on the page.

**Comment by arnieswap** [comment]
I tried to install some packages and now I get this error:

Conversion of The_Iron_Lady.mp4 to Generic Android Droid X failed! Reason: Unable to construct pipeline! no property "profile" in element "faac0"

**Comment by danielgtaylor** [comment]
That is a GStreamer issue - you should file a bug with OpenSUSE's GStreamer
package. It's possible it is out of date or has some non-standard patches
applied. You may refer to this bug when you file one there so they know the
context.

On Tue, Mar 27, 2012 at 08:07, arnieswap <
reply@reply.github.com

> wrote:
> 
> I tried to install some packages and now I get this error:
> 
> Conversion of The_Iron_Lady.mp4 to Generic Android Droid X failed! Reason:
> Unable to construct pipeline! no property "profile" in element "faac0"
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/danielgtaylor/arista/issues/134#issuecomment-4714816

## 

Daniel G. Taylor
http://programmer-art.org/

**Comment by arnieswap** [comment]
Thanks I already filed a bug with them. 
https://bugzilla.novell.com/show_bug.cgi?id=753234
Thanks.
Swapnil

**Comment by reddwarf69** [comment]
gst-inspect faac indeed shows that there is no profile property in the faac element. But I neither see it in the code: http://cgit.freedesktop.org/gstreamer/gst-plugins-bad/tree/ext/faac/gstfaac.c

It looks like a problem in arista (or the profile), not in the openSUSE packages.

**Comment by danielgtaylor** [comment]
It is there, take a look at e.g. lines 85, 91, 108, 415, 441-451, 598-617,
etc.

On Tue, Apr 3, 2012 at 06:34, RedDwarf69 <
reply@reply.github.com

> wrote:
> 
> gst-inspect faac indeed shows that there is no profile property in the
> faac element. But I neither see it in the code:
> http://cgit.freedesktop.org/gstreamer/gst-plugins-bad/tree/ext/faac/gstfaac.c
> 
> It looks like a problem in arista (or the profile), not in the openSUSE
> packages.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/danielgtaylor/arista/issues/134#issuecomment-4901251

## 

Daniel G. Taylor
http://programmer-art.org/

**Comment by reddwarf69** [comment]
OK, it's a property of the source pad, I didn't look there. But in such a case a gst-inspect shows the element provides it.

It's really difficult to me to see anything wrong in the packages. Can you confirm that you can use the "Droid X" preset with GStreamer 0.10.36 and gst-plugins-bad 0.10.23?

Since I don't speak Python it would be a bit difficult to me to debug this. If it works for you, can you give a "gst-launch" line equivalent to what arista does when using the Droid X preset?

**Comment by reddwarf69** [comment]
OK, I actually looked in the correct place before. The problem is this commit: http://cgit.freedesktop.org/gstreamer/gst-plugins-bad/commit/ext/faac/gstfaac.c?id=6702de4237f6e7c3166b2807150669557220c44f
Arista doesn't work with the latest version of the faac element.

I don't really know GStreamer so well to say if this is a bug or not. But you should either stop depending on the faac element providing that property or open a bug report with GStreamer for breaking your app.

**Comment by reddwarf69** [comment]
To finish it. 0.10.23 was the first version of gst-plugins-bad to include this change. And to make it work, instead of "faac bitrate=<bitrate> profile=LC", "faac bitrate=<bitrate> ! audio/mpeg, base-profile=lc" must be used.

**Comment by danielgtaylor** [comment]
Interesting. For now you can manually update your preset .json file with
that new string and things should work for you.

In the future, I'll have to detect the version and act accordingly, which
means probably overriding or transforming the presets... this will be kind
of a pain.

On Tue, Apr 3, 2012 at 08:30, RedDwarf69 <
reply@reply.github.com

> wrote:
> 
> To finish it. 0.10.23 was the first version of gst-plugins-bad to include
> this change. And to make it work, instead of "faac bitrate=<bitrate>
> profile=LC", "faac bitrate=<bitrate> ! audio/mpeg, base-profile=lc" must be
> used.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/danielgtaylor/arista/issues/134#issuecomment-4904047

## 

Daniel G. Taylor
http://programmer-art.org/

**Comment by arnieswap** [comment]
Thanks that solved the problem and now I am able to transcode videos. I will write about it to help other users as well. Thanks again.

**Comment by kapouer** [comment]
Actually i had to just remove profile key, since base-profile doesn't work either here on debian/sid + deb-multimedia.
